presto-main/src/main/java/io/prestosql/server/security/UserMapping.java

```diff
@@ -84,33 +86,57 @@ public UserMappingRules(
         }
     }
 
+    enum Case
+    {
+        KEEP(Function.identity()),
+        LOWER(value -> value.toLowerCase(Locale.ENGLISH)),
+        UPPER(value -> value.toUpperCase(Locale.ENGLISH)),
+        /**/;
+
+        private final Function<String, String> transformation;
+
+        Case(Function<String, String> transformation)
```

Comment:
Consider the simpler (albeit slightly longer) pattern of implementing a method (the IDE folds it nicely)
```java
enum Case
{
    KEEP {
        @Override
        public String transform(String value)
        {
            return value;
        }
    },
    LOWER {
        @Override
        public String transform(String value)
        {
            return value.toLowerCase(Locale.ENGLISH);
        }
    },
    UPPER {
        @Override
        public String transform(String value)
        {
            return value.toLowerCase(Locale.ENGLISH);
        }
    };

    public abstract String transform(String value);
}

Reply:
done